ompi/mca/common/ompio/common_ompio_file_open.c

```diff
@@ -480,26 +481,76 @@ int mca_common_ompio_file_delete (const char *filename,
                                   struct opal_info_t *info)
 {
     int ret = OMPI_SUCCESS;
+    ompio_file_t *fh = NULL;
 
     /* No locking required for file_delete according to my understanding.
        One thread will succeed, the other ones silently ignore the 
        error that the file is already deleted.
     */
-    ret = unlink(filename);
-
-    if (0 > ret ) {
-        if ( ENOENT == errno ) {
-            return MPI_ERR_NO_SUCH_FILE;
-        } else {
-            opal_output (0, "mca_common_ompio_file_delete: Could not remove file %s errno = %d %s\n", filename,
-                         errno, strerror(errno));
-            return MPI_ERR_ACCESS;
-        }
+
+    /* Create an incomplete file handle, it will basically only
+       contain the filename. It is needed to select the correct
+       component in the fs framework and call the file_remove
+       function corresponding to the file type. 
+    */
+    ret = mca_common_ompio_create_incomplete_file_handle(filename, &fh);
+    if (ret != OMPI_SUCCESS) {
+        return ret;
+    }
+
+    ret = mca_fs_base_file_select (fh, NULL);
+    /* We don't need the file handle anymore */
+    free(fh);
+    if (OMPI_SUCCESS != ret) {
+        opal_output(1, "mca_fs_base_file_select() failed\n");
+        return ret;
+    }
+
+    ret = fh->f_fs->fs_file_delete (filename, NULL);
+    if (ret != OMPI_SUCCESS) {
```

Comment:
Please move the free(fh) statement down, after the the line where you are de-referencing fh in line 509.
Otherwise looks good.